include/jemalloc/internal/prof_inlines_a.h

```diff
@@ -18,33 +43,32 @@ prof_accum_add(tsdn_t *tsdn, prof_accum_t *prof_accum,
 	 * avoids rate-limiting allocation.
 	 */
 #ifdef JEMALLOC_ATOMIC_U64
-	a0 = atomic_load_u64(&prof_accum->accumbytes, ATOMIC_RELAXED);
+	a0 = atomic_load_u64(&prof_accum.accumbytes, ATOMIC_RELAXED);
 	do {
 		a1 = a0 + accumbytes;
 		assert(a1 >= a0);
 		overflow = (a1 >= prof_interval);
 		if (overflow) {
 			a1 %= prof_interval;
 		}
-	} while (!atomic_compare_exchange_weak_u64(&prof_accum->accumbytes, &a0,
+	} while (!atomic_compare_exchange_weak_u64(&prof_accum.accumbytes, &a0,
 	    a1, ATOMIC_RELAXED, ATOMIC_RELAXED));
 #else
-	malloc_mutex_lock(tsdn, &prof_accum->mtx);
-	a0 = prof_accum->accumbytes;
+	malloc_mutex_lock(tsdn, &prof_accum.mtx);
+	a0 = prof_accum.accumbytes;
 	a1 = a0 + accumbytes;
 	overflow = (a1 >= prof_interval);
 	if (overflow) {
 		a1 %= prof_interval;
 	}
-	prof_accum->accumbytes = a1;
-	malloc_mutex_unlock(tsdn, &prof_accum->mtx);
+	prof_accum.accumbytes = a1;
+	malloc_mutex_unlock(tsdn, &prof_accum.mtx);
 #endif
 	return overflow;
 }
 
 static inline void
-prof_accum_cancel(tsdn_t *tsdn, prof_accum_t *prof_accum,
-    size_t usize) {
+prof_idump_rollback(tsdn_t *tsdn, size_t usize) {
```

Comment:
Similarly, add an early bail with `prof_interval == 0 || !prof_active_get_unlocked()`, and make the body non inline.

Reply:
Same question as above.